core/src/main/java/hudson/FilePath.java

```diff
@@ -587,11 +588,11 @@ private void unzip(File dir, File zipFile) throws IOException {
                 ZipEntry e = entries.nextElement();
                 File f = new File(dir, e.getName());
                 if (e.isDirectory()) {
-                    mkdirs(f);
+                    Files.createDirectories(f.toPath());
```

Comment:
Use `Util#toPath()` created by @dwnusbaum a while ago. Otherwise there is a risk of unhandled runtime `InvalidPathException`

Reply:
Also `FilePath#mkdirs(File)` calls `filterNonNull().mkdirs(dir);`, which checks permissions when doing remote operations. This removes that check. I would update the `mkdirs` method itself to use `Files#createDirectories`, and then you won't need to change all of its callers.